screen_test.go

```diff
@@ -0,0 +1,55 @@
+package terminal
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+var currentLineForWritingTestCases = []struct {
+	name     string
+	input    string
+	want     []string
+	maxlines int
+}{
+	{
+		name: "Test no index out of range panic",
+		input: "\n",
+		want: []string{"&nbsp;\n"},
+		maxlines: 1,
+	},
+	{
+		name: "Test scroll out first line",
+		input: "a\n",
+		want: []string{"a\n"},
+		maxlines: 1,
+	},
+	{
+		name: "Test scroll out several lines",
+		input: "a\nb\nc\nd",
+		want: []string{"a\n", "b\n"},
+		maxlines: 2,
+	},
+}
+
+func TestCurrentLineForWriting(t *testing.T) {
+	for _, test := range currentLineForWritingTestCases {
+		s, err := NewScreen(WithMaxSize(0, test.maxlines))
+		got := []string{}
+		s.ScrollOutFunc = func(line string) { got = append(got, line) }
+		_ = s.currentLineForWriting()
+		s.Write([]byte(test.input))
+		if err != nil {
+			t.Errorf("Failure for '%s':\nNewScreen returned an error: %s", test.name, err)
+		}
+		_ = s.currentLineForWriting()
+
+		if diff := cmp.Diff(got, test.want); diff != "" {
+			t.Errorf(
+				"Failure for '%s':\nscrolledOutFunc sequence of parameters diff (-got +want):\n%s",
+				test.name,
+				diff,
+			)
+		}
+	}
```

Comment:
Wrapping the loop body in `t.Run` would be useful for a few reasons, such as not needing to include the testcase name within each `t.Errorf` call.

The other thing I would change is to move the `NewScreen` error check up, include its argument in the log, and (now that it's in a `t.Run`) change that particular failure from `t.Errorf` to `t.Fatalf`.

```suggestion
	for _, test := range currentLineForWritingTestCases {
		t.Run(test.name, func(t *testing.T) {
			s, err := NewScreen(WithMaxSize(0, test.maxlines))
			if err != nil {
				t.Fatalf("NewScreen(WithMaxSize(0, %d)) error: %s", test.maxlines, err)
			}
			got := []string{}
			s.ScrollOutFunc = func(line string) { got = append(got, line) }
			_ = s.currentLineForWriting()
			s.Write([]byte(test.input))
			_ = s.currentLineForWriting()

			if diff := cmp.Diff(got, test.want); diff != "" {
				t.Errorf("scrolledOutFunc sequence of parameters diff (-got +want):\n%s", diff)
			}
		})
	}
```